### PR TITLE
feat(curriculum): adding content for recursion review page

### PR DESCRIPTION
**Pull request by zairahira**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/i18n/locales/english/intro.json [hunk 1]
```diff
@@ -3167,7 +3167,8 @@
       "review-recursion": {
         "title": "Recursion Review",
         "intro": [
-          "Review the Recursion concepts to prepare for the upcoming quiz."
+          "Before you are quizzed on the recursion, you first need to review the concepts.",
+          "Open up this page to review what is recursion and what is it used for"
```
Comment (jdwilkin4) [inline client/i18n/locales/english/intro.json:3171]:
```suggestion
          "Open up this page to review what is recursion and what is it used for."
```

File: curriculum/challenges/english/25-front-end-development/review-recursion/6723d1f0568292cd394d6fb6.md [hunk 2]
```diff
@@ -9,7 +9,10 @@ dashedName: review-recursion
 
 Review the concepts below to prepare for the upcoming quiz.
 
-
+- Recursion is programming concept that allows you to call a function repeatedly until a base-case is reached.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/review-recursion/6723d1f0568292cd394d6fb6.md:12]: I would include an example here

**Review by jdwilkin4 — COMMENTED** [review]
Left a couple of comments 👍

**Review by huyenltnguyen — COMMENTED** [review]
I gave this a quick glance because I'm helping with the recursion quiz 😄 

Since the lecture also covers call stack, I think we probably want to include that in the review?

**Review by jdwilkin4 — APPROVED** [review]

**Review by moT01 — APPROVED** [review]

**Pull request merged**